s2i-liberty/readme.md

```diff
@@ -0,0 +1,197 @@
+* `~` - Specifying that source code will be provided
+* `https://github.com/redhat-cop/containers-quickstarts` - URL of the Git repository
+* `--context-dir` - Location within the repository containing source code
+* `--name=s2i-liberty` - Name for the build and resulting image
+* `--strategy=docker` - Name of the OpenShift source strategy that is used to produce the new image
+
+*Note: If the repository was moved to a different location (such as a fork), be sure to reference to correct location.*
+
+A new image called *s2i-liberty* was produced and can be used to build Liberty applications in the subsequent sections.
+
+#### Binary Source
+
+Instead of referencing a git repository, the content can be provided directly to the OpenShift build process using a binary source build. 
+
+The first step is to obtain the source code containing the builder. Once the code has been obtained, navigate to the folder containing the Play Framework *Dockerfile*, and execute the following command to start a new image build using the binary source strategy:
+
+```
+oc new-build websphere-liberty:webProfile7 --name=s2i-liberty --strategy=docker --from-dir=. --binary=true
+```
+
+Let's break down the command in further detail
+
+* `oc new-build` - OpenShift command to create a new build
+* `websphere-liberty:webProfile7` - The location of the base Docker image for which a new ImageStream will be created
+* `--from-dir` - Location of source code that will be used as the source for the build process. The contents will be tar'ed up and uploaded to the builder image.
+* `--name=s2i-liberty` - Name for the build and resulting image
+* `--strategy=docker` - Name of the OpenShift source strategy that is used to produce the new image
+* `--binary=true` - Specifies this build will be of a binary source type
+
+A new image called *s2i-liberty* was produced and can be used to build Play Framework applications in the subsequent sections.
+
+### Create the build config
+
+In order to use the extended s2i process we need to configure the build config with some additional pieces of information. There does not appear to be a way to do this directly from the command line so we will use a two-phase approach.
+
+First we will create a standard build config using the git source apporach explained above. Run the follwoing command:
+```
+oc new-build --docker-image=registry.access.redhat.com/jboss-eap-7/eap70-openshift --code=https://github.com/efsavage/hello-world-war --name=hello-world
+```
+Second modify the newly created build config to look as follows:
+```
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: 'eap70-openshift:latest'
+      runtimeImage:
+        kind: ImageStreamTag
+        name: 'liberty-runtime-s2i:latest'
+      runtimeArtifacts:
+        - sourcePath: /opt/eap/standalone/deployments/hello-world-war-1.0.0.war
+          destinationDir: artifacts
+```
+notice that the source strategy has two additional sections:
```

Comment:
Can we provide this modifications as a _patch_ command that users can easily apply?